lib/_stream_writable.js

```diff
@@ -421,27 +421,24 @@ function onwrite(stream, er) {
       onwriteError(stream, state, er, cb);
     }
   } else {
-    if (!state.destroyed) {
+    if (state.buffered.length > state.bufferedIndex) {
```

Comment:
since `clearBuffer` is not inlineable this branch remove a lot of the overhead of calling into `clearBuffer`